edb/server/compiler/sertypes.py

```diff
@@ -233,9 +233,9 @@ def _describe_type(self, t, view_shapes, view_shapes_metadata,
                 links.append(not ptr.is_property(self.schema))
 
             t_rptr = t.get_rptr(self.schema)
-            if t_rptr is not None:
+            if t_rptr is not None and (rptr_ptrs := view_shapes.get(t_rptr)):
```

Comment:
I'm not sure this is a correct fix... So there can be a type that has a shape but for some reason the type descriptor for it would be empty? If you believe that the fix is correct then add a detailed comment explaining why (more detailed than what's in the commit message).

Reply:
It's not a type, it's a link, which would only have a shape if you've got link properties in your shape.  The reason why this code mostly works is because we use `defaultdict` in the compiler for the `view_shapes` map.  Other references to `view_shapes` in `sertypes` use `.get()` appropriately.

Reply:
> It's not a type, it's a link

Ah, I see now